neuro_evolution_ctrnn/batch_generate_plot.py

```diff
@@ -1,31 +1,38 @@
 import os
 import logging
-import argparse
 import subprocess
+from tap import Tap
 
 logging.basicConfig(format='%(levelname)s:%(message)s', level=logging.INFO)
 
-parser = argparse.ArgumentParser(
-    description='iterate over simulation folders and create a plot for each simulation run')
 
-parser.add_argument('--base-dir', metavar='dir', type=str,
-                    help='path to the parent-folder of simulation results',
-                    default=os.path.join('..', 'CTRNN_Simulation_Results', 'data'))
+class BatchPlotArgs(Tap):
+    # Note: Comments in lines become helpstring when called with --help
+    # for details on parsing see: https://github.com/swansonk14/typed-argument-parser
 
-parser.add_argument('--plot-file-name', metavar='filename', type=str,
-                    help='the filename under which to store the plots',
-                    default='plot.png')
+    basedir: str  # path to the parent-folder of simulation results
+    filename: str = "plot.png"  # the filename under which to store the plots
+    handle_existing: str = "skip"  # what to do when the plot-file already exists? options: skip, replace, raise
+    smooth: int = 0  # How strong should the lines be smoothed? (0 to disable)
 
-parser.add_argument('--handle-existing', type=str,
-                    help='what to do when the plot-file already exists? options: skip, replace, raise',
-                    default='skip')
+    def configure(self):
+        self.description = "iterate over simulation folders and create a plot for each simulation run by" \
+                           "by calling plot_experiment.py for each folder"
 
-args = parser.parse_args()
+        # positional argument:
+        # note: for reasons unknown positional arguments can not contain underscores when using underscores_to_dashes
+        self.add_argument("basedir")
 
-simulation_folders = [f.path for f in os.scandir(args.base_dir) if f.is_dir()]
+        # aliases
+        self.add_argument("-e", "--handle_existing")
+
+
+args = BatchPlotArgs(underscores_to_dashes=True).parse_args()
+
+simulation_folders = [f.path for f in os.scandir(args.basedir) if f.is_dir()]
```

Comment:
```suggestion
simulation_folders = [f.path for f in os.scandir(args.basedir)]
```

Das if kann weg denke ich -> kriege so oder so ein error wenn args.basedir kein Folder ist

Reply:
ich denke das ist da, für so einen fall

/base_dir/simulationresult1/
/base_dir/simulationresult2/
/base_dir/simulationresult2/
/base_dir/file.txt

Da soll die file.txt dann keinen fehler werfen